### ISSUE TITLE
LLVM ERROR: character buffer should be in CharBoxValue

### ISSUE THREAD

**Issue by sscalpone** [flang-compiler/f18-llvm-project#495]
From sixtrack

```
% cat dv1.f
          SUBROUTINE KERSET(ERCODE)
          CHARACTER*6         ERCODE,   CODE(28)
          DATA      CODE(1)  / 'C204.1' /
          RETURN
          ENTRY KERMTR(ERCODE)
          ERCODE = CODE(1)
          RETURN
          END
```
```
% bbc dv1.f |& head
LLVM ERROR: character buffer should be in CharBoxValue
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace.
Stack dump:
0.      Program arguments: bbc dv1.f
 #0 0x0000562140b9550d llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/local/home/sjs/mirror/install/bin/bbc+0x340f50d)
 #1 0x0000562140b93244 llvm::sys::RunSignalHandlers() (/local/home/sjs/mirror/install/bin/bbc+0x340d244)
 #2 0x0000562140b933b0 SignalHandler(int) (/local/home/sjs/mirror/install/bin/bbc+0x340d3b0)
 #3 0x000014f0a26ec8a0 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x128a0)
 #4 0x000014f0a1358f47 raise /build/glibc-2ORdQG/glibc-2.27/signal/../sysdeps/unix/sysv/linux/raise.c:51:0
 #5 0x000014f0a135a8b1 abort /build/glibc-2ORdQG/glibc-2.27/stdlib/abort.c:81:0
```

**Comment by jeanPerier** [comment]
Issue comes from alternate entry SymbolBox copy that is not copying all the SymbolBox content, but only the address:
https://github.com/flang-compiler/f18-llvm-project/blob/aa370cad69b060d7a75d0aa262765399c4f49352/flang/lib/Lower/Bridge.cpp#L1742

I am working on a fix.

**Comment by jeanPerier** [comment]
Should be fixed by #501 that was merged.

**Comment by sscalpone** [comment]
Verified.
